LambdaEngine/Include/Resources/Mesh.h

```diff
@@ -61,29 +50,50 @@ namespace LambdaEngine
 		uint8 Padding;
 	};
 
-	// Moved out from mesh due to dependency issue
-	using MeshIndexType = uint32;
-	
+	using MeshIndexType = uint32; // Moved out from mesh due to dependency issue
+	using JointIndexType = uint8; // Limited to 128 bones, enough for now
+
+	// Animation
+	struct VertexJointData
+	{
+		JointIndexType JointID0 = INVALID_JOINT_ID;
+		JointIndexType JointID1 = INVALID_JOINT_ID;
+		JointIndexType JointID2 = INVALID_JOINT_ID;
+		JointIndexType JointID3 = INVALID_JOINT_ID;
+		float32 Weight0;
+		float32 Weight1;
+		float32 Weight2;
+		// Last weight is calculated in shader since they must equal 1.0
+	};
+
 	struct VertexWeight
 	{
 		MeshIndexType	VertexIndex;
 		float32			VertexWeight;
 	};
 
-	struct Bone
+	struct Joint
 	{
-		PrehashedString			Name;
-		glm::mat4				OffsetTransform;
-		TArray<VertexWeight>	Weights;
-		int32					ParentBoneIndex = -1;
+		// TODO: Change into mat4x3
```

Comment:
Is this something you've forgotten to do, or you're planning to do further ahead?

Reply:
Future change, mat4 is quite deeply integrated into the animation pipeline right now. For now a note to self.